planning/scenario_planning/parking/astar_search/src/astar_search.cpp

```diff
@@ -346,8 +350,8 @@ bool AstarSearch::search()
 
 void AstarSearch::setPath(const AstarNode & goal_node)
 {
-  std_msgs::Header header;
-  header.stamp = ros::Time::now();
+  std_msgs::msg::Header header;
+  header.stamp = rclcpp::Clock().now();
```

Comment:
This is not a rosnode, so `this->get_clock()` is not an option. I'm checking the difference between `rclcpp::Node::now()` and `rclcpp::Clock().now()`.

Reply:
@jilaada In this case (no Nodes are defined), I cannot see any differences between `rclcpp::Node::now()` and `rclcpp::Clock().now()`. How do you think?

http://docs.ros2.org/foxy/api/rclcpp/classrclcpp_1_1Node.html#a1db9cec1af19a994b3f1dad2059d2707

Reply:
@TakaHoribe Here is some reference issue I found regarding the difference between Node and Clock time: https://github.com/ros2/rclcpp/issues/595. It seems like Node time allows for the use of sim time when use sim time is enabled. It seems like Clock only takes the the system time. But since this is not a node, I'm not sure what the behaviour will be? :thinking:

Reply:
I never checked but I thought it should be able to refer sim time if you give RCL_ROS_TIME as constructor.
http://docs.ros2.org/dashing/api/rcl/time_8h.html#a5c734f508ce06aec7974af10ad09d071

i.e. 
rclcpp::Clock(RCL_ROS_TIME).now();

Reply:
Yeah this would be the best option I think. The default clock implementation uses the `RCL_SYSTEM_TIME` which would not be simulator friendly: https://github.com/ros2/rclcpp/blob/master/rclcpp/include/rclcpp/clock.hpp#L66